webserver/webserver/src/main/java/io/helidon/webserver/ForwardingHandler.java

```diff
@@ -178,14 +198,49 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) {
             }
 
             if (msg instanceof LastHttpContent) {
-                requestContext.publisher().complete();
-                requestContext = null; // just to be sure that current http req/res session doesn't interfere with other ones
+                if (!isWebSocketUpgrade) {
+                    requestContext.publisher().complete();
+                    requestContext = null; // just to be sure that current http req/res session doesn't interfere with other ones
+                }
             } else if (!content.isReadable()) {
                 // this is here to handle the case when the content is not readable but we didn't
                 // exceptionally complete the publisher and close the connection
                 throw new IllegalStateException("It is not expected to not have readable content.");
             }
         }
+
+        // We receive a raw bytebuf if connection was upgraded to WebSockets
+        if (msg instanceof ByteBuf) {
+            if (!isWebSocketUpgrade) {
+                throw new IllegalStateException("Received ByteBuf without upgrading to WebSockets");
+            }
+            // Simply forward raw bytebuf to Tyrus for processing
+            LOGGER.fine("Received ByteBuf of WebSockets connection" + msg);
```

Comment:
This looks like a candidate for `finest`, not `fine`.
Also when constructing a message in log levels lower than `info`, please use lambda:
`LOGGER.finest(() -> "..."  + msg);` to avoid string concatenation on every single byte buffer received